web_client/analysis-modules/igpse-lib.jsx

```diff
@@ -0,0 +1,619 @@
+
+import ItemModel from 'girder/models/ItemModel';
+
+import actions, { addItemMetadata } from '../actions';
+import analysisUtils from '../utils/analysis';
+import { Promise, delayedSemaphore } from '../utils/promise';
+import { store } from '../globals';
+import loadModel from '../utils/load-model';
+
+const workflowTable = {
+  input: require('./igpse-input'),
+  featureSelection: require('./igpse-feature-selection'),
+  subsetSelection: require('./igpse-subset-selection'),
+  subsetSelectionB: require('./igpse-subset-selection-b'),
+  survivalPlot: require('./igpse-survival-plot')
+};
+
+const D = store.dispatch.bind(store);
+
+const noop = () => {};
+
+export class IGPSEWorkflow {
+  constructor() {
+    this.elementObjects = {};
+    this.elements = {};
+    this.pages = {};
+    this.index = 0;
+
+    this.workflowTableCache = {};
+
+    this.constructPromise = Promise.mapSeries(
+      [
+        ['igpse-input'             , 'input'           , true , true],
+        ['igpse-feature-selection' , 'featureSelection', true , true],
+        ['igpse-subset-selection-a', 'subsetSelection' , true , true],
+        ['igpse-subset-selection-b', 'subsetSelectionB', false, true],
+        ['igpse-survival-plot'     , 'survivalPlot'    , true , true]
+      ],
+      (args) => this.processPage(...args)
+    );
+
+    window.G = this;
```

Comment:
But I gotta ask: the table above this line should have been "fixed" per the linter's suggestion that the extra whitespace be removed.  I'm not seeing it reflected in this version of the file?